### PR TITLE
Add case insensitive comparison for contains operator

### PR DESCRIPTION
**Pull request by FrankHeijden**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ Add the repo to your project inside your `pom.xml` file
 ```
 
 ##### Gradle
-Add the repo to your project by adding `mavenLocal()` to `repositories` and `implementation 'com.sipios:spring-search:0.2.0'` to  `dependencies` in your `build.gradle` file.
```
Comment (wrousseau) [inline README.md:99]: Why did you remove this instruction?

File: src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt [hunk 2]
```diff
@@ -24,13 +24,16 @@ interface ParsingStrategy {
             SearchOperation.NOT_EQUALS -> builder.notEqual(path.get<Any>(fieldName), value)
             SearchOperation.STARTS_WITH -> builder.like(path.get(fieldName), "$value%")
             SearchOperation.ENDS_WITH -> builder.like(path.get(fieldName), "%$value")
-            SearchOperation.CONTAINS -> builder.like((path.get<String>(fieldName).`as`(String::class.java)), "%$value%")
+            SearchOperation.CONTAINS -> {
```
Comment (wrousseau) [inline src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt:27]: Not a big fan of this as it will be a breaking change. Would rather add a ICONTAINS operator, what do you think?

Comment (wrousseau) [reply src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt:27]: Also we need to adapt the tests for this

**Comment by wrousseau** [comment]
@FrankHeijden thanks a lot for the Pull Request, sorry for reviewing it so late. i've made a couple of comments

File: README.md (hunk 1 above)
Comment (FrankHeijden) [reply README.md:99]: I did not make this change, it’s a previous commit which has been done to master, I made changes to the `feature/add-more-types` and merged it with master in my local repo.

File: src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt (hunk 2 above)
Comment (FrankHeijden) [reply src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt:27]: I don’t think it will be a breaking change as the branch im merging with is not yet public; but we could have a different searchoperator which distinguishes between exact matches and ignorecase.

Comment (FrankHeijden) [reply src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt:27]: Also I think the tests work fine for it right now, they don't give any errors.

**Comment by FrankHeijden** [comment]
Closing old PR
